code/modules/mining/equipment/kheiral_cuffs.dm

```diff
@@ -0,0 +1,120 @@
+/**********************Kheiral Cuffs**********************/
+/// Acts as a GPS beacon & distress signaller when clamped to one's wrist off-station.
+/obj/item/kheiral_cuffs
+	name = "\improper Kheiral cuffs"
+	desc = "A prototype wrist communicator powered by Kheiral Matter. When both ends are clamped to one wrist, can send a distress signal upon your death off-station.\nA small engraving on the inside reads, \"NOT HANDCUFFS\""
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "strand"
+	worn_icon_state = "strandcuff"
+	slot_flags = ITEM_SLOT_GLOVES
+	throwforce = 0
+	w_class = WEIGHT_CLASS_SMALL
+	gender = PLURAL
+	throw_speed = 3
+	throw_range = 5
+	/// If we're in the glove slot
+	var/on_wrist = FALSE
+	/// If the GPS is already on
+	var/gps_enabled = FALSE
+	/// If we're off the station's Z-level
+	var/far_from_home = FALSE
+	/// Used to broadcast user's death through the radio
+	var/obj/item/radio/internal_radio
+
+/obj/item/kheiral_cuffs/Initialize()
+	. = ..()
+	update_icon(UPDATE_OVERLAYS)
+	RegisterSignal(src, COMSIG_MOVABLE_Z_CHANGED, .proc/check_z)
+	internal_radio = new/obj/item/radio(src)
+	internal_radio.subspace_transmission = TRUE
+	internal_radio.canhear_range = 0
+	internal_radio.should_be_listening = FALSE
+	internal_radio.keyslot = new /obj/item/encryptionkey/strandcuffs
+	internal_radio.set_listening(FALSE)
+	internal_radio.recalculateChannels()
+
+/obj/item/kheiral_cuffs/examine(mob/user)
+	. = ..()
+	if(gps_enabled)
+		. += span_notice("The cuff's GPS signal is on.")
+
+/obj/item/kheiral_cuffs/item_action_slot_check(slot)
+	return slot == ITEM_SLOT_GLOVES
+
+/obj/item/kheiral_cuffs/equipped(mob/user, slot, initial)
+	. = ..()
+	if(slot != ITEM_SLOT_GLOVES)
+		return
+	on_wrist = TRUE
+	playsound(loc, 'sound/weapons/handcuffs.ogg', 30, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	connect_kheiral_network(user)
+
+/obj/item/kheiral_cuffs/dropped(mob/user, silent)
+	. = ..()
+	on_wrist = FALSE
+	playsound(loc, 'sound/weapons/handcuffs.ogg', 30, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	remove_kheiral_network(user)
+
+/// Enables the GPS & registers the statchange signal
+/obj/item/kheiral_cuffs/proc/connect_kheiral_network(mob/user)
+	if(gps_enabled)
+		return
+	if(on_wrist && far_from_home)
+		AddComponent(/datum/component/gps, "*[user.name]'s Kheiral Link")
+		balloon_alert(user, "GPS activated")
+		RegisterSignal(user, COMSIG_MOB_STATCHANGE, .proc/on_user_statchange)
+		gps_enabled = TRUE
+
+/// Disables the GPS and unregisters the statchange signal
+/obj/item/kheiral_cuffs/proc/remove_kheiral_network(mob/user)
+	if(!gps_enabled)
+		return
+	if(!on_wrist || !far_from_home)
+		qdel(GetComponent(/datum/component/gps))
+		balloon_alert(user, "GPS de-activated")
+		UnregisterSignal(user, COMSIG_MOB_STATCHANGE)
+		gps_enabled = FALSE
+
+/obj/item/kheiral_cuffs/Destroy(force)
+	. = ..()
+	UnregisterSignal(src, COMSIG_MOVABLE_Z_CHANGED)
+	if(internal_radio)
+		qdel(internal_radio)
```

Comment:
This will hard delete the radio won't it?

Reply:
I'm not really sure!
I should have really looked to see if anything else did this too.

Reply:
I forgot to make it QDEL_NULL when I made it, I fixed that in that last commit I did. Now it should be the same as how bots do it.